lib/packages/docutils/rst.nim

```diff
@@ -2904,6 +2970,63 @@ proc preparePass2*(s: PRstSharedState, mainNode: PRstNode) =
   countTitles(s, mainNode)
   orderFootnotes(s)
 
+proc normalizeNimName*(s: string): string =
+  ## Makes `s` lower case except the first character in any symbol,
+  ## deletes underscores, removes spaces before and after punctuation
+  ## characters, removes Nim export symbol (*).
+  ## It assumes that spaces cannot be repetitive in the input string `s`
+  ## (because the previous stages of parser has already turned multiple
+  ## spaces or a newline inside a link `s` into 1 space).
+  runnableExamples:
+    let ex1 = "proc binarysearch(T,K,proc(T,K))"
+    doAssert  "proc binarySearch(T, K, proc (T, K))".normalizeNimName == ex1
+    doAssert ("proc binarySearch[T, K] (T, K, proc (T, K))".normalizeNimName ==
+              ex1)
+    let ex2 = "iterator returnall(openarray[T])"
+    doAssert ("iterator returnAll [ T ] ( openArray[T] )".normalizeNimName ==
+              ex2)
+    doAssert ("iterator returnAll ( openArray[T] )".normalizeNimName ==
+              ex2)
+    doAssert ("iterator return_all* ( open_array[T] )".normalizeNimName ==
+              ex2)
+    doAssert  "type SortOrder".normalizeNimName == "type Sortorder"
+    doAssert  "SortOrder".normalizeNimName == "Sortorder"
+  var i = 0
+  var newWord = true
+  var afterSymbolName = true
+  const nimDeclPunctuation = {',', ';', '(', ')', '[', ']', '*', '{', '}'}
+  while i < s.len:
+    case s[i]
+    of ' ':
+      if (i > 0 and s[i-1] in nimDeclPunctuation) or
+          (i+1 < s.len and s[i+1] in nimDeclPunctuation):
+        inc i
+      else:
+        result.add s[i]
+        inc i
+      newWord = true
+    of nimDeclPunctuation:
+      if afterSymbolName and s[i] == '[':
+        while i < s.len:  # skip type parameters in brackets
```

Comment:
er... what? `proc f(x: G[int])` is not the same as `proc f(x: G[string])`

Reply:
it's just a bad wording, I changed it. I mean only "definition" brackets right after symbol name and before opening `(`.

I also added test case to `utils.nim` to ensure it works fine.